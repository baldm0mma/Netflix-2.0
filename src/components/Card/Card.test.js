import {
  Card,
  mapStateToProps,
  mapDispatchToProps
} from './Card';
import {
  shallow
} from 'enzyme';
import React from 'react';

describe('Card', () => {
  describe('component', () => {
    let wrapper, instance;

    beforeEach(() => {
      wrapper = shallow( < Card / > );
      instance = wrapper.instance()
    });
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should invoke seeSpecficMovie on click', () => {
      expect(wrapper.state('clicked')).toEqual(false);
      wrapper.find('.view-details-btn').simulate('click');
      instance.seeSpecificMovie();
      expect(wrapper.state('clicked')).toEqual(true);
    });
    it('should invoke sendFavorite on click', () => {
      expect(wrapper.state('clicked')).toEqual(false);
      wrapper.find('.favorite-movie-btn').simulate('click');
      instance.sendFavorite();
      expect(wrapper.state('clicked')).toEqual(true);
    });
  });
  describe('mapStateToProps', () => {
    it('should return an object with the user and specificMovie', () => {

    });
  });
  describe('mapDispatchToProps', () => {
    it('should', () => {

    });
  });
});