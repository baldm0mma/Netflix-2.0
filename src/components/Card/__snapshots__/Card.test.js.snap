// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot 1`] = `undefined`;

exports[`Card should match snapshot 1`] = `undefined`;
